app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ShoppingListController < ApplicationController
+  before_action :authenticate_user!
+
+  def sum(array)
+    sum = 0
+    array.each do |number|
+      sum += number.quantity * number.food.price
+    end
+    sum
+  end
+
+  def index
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_foods = RecipeFood.all
+    @food = Food.all
+    @total_price = sum(@recipe.recipe_foods.includes([:food]))
+  end
```

Comment:
```suggestion
  def index
    @recipe = Recipe.find(params[:recipe_id])
    @recipe_foods = RecipeFood.all
    @food = Food.all
    @total_price = sum(@recipe.recipe_foods.includes([:food]))
  end

  private
  
  def sum(array)
    sum = 0
    array.each do |number|
      sum += number.quantity * number.food.price
    end
    sum
  end
```

Reply:
- [optional] I think it will be better and cleaner, If we make the `sum` method private